wikimedia/wikipedia-ios
# Integrate Watchlist Filter Component

**Phabricator:**
https://phabricator.wikimedia.org/T342929
https://phabricator.wikimedia.org/T335817 (partially)

### Notes
This PR instantiates the watchlist filter view model (with the proper localized strings) established in https://github.com/wikimedia/wikipedia-ios-components/pull/21. This PR also creates a `SharedContainerCacheStore`, which the data framework will call when it attempts to fall back to offline mode.

This PR is dependent on https://github.com/wikimedia/wikipedia-ios-components/pull/21 & https://github.com/wikimedia/wikipedia-ios-data/pull/6. After those are merged, please update the Components package revision ID to the merge commit off of the Components main branch, then merge this one.

### Test Steps
1. Temporarily add this code to WKWatchlistViewModel and WKWatchlistView:

WKWatchlistViewModel:
```
    private let dataController = WKWatchlistDataController()

    func fetchWatchlist() {
        dataController.fetchWatchlist { result in
            switch result {
            case .success(let watchlist):
                print(watchlist)
            case .failure(let error):
                print(error)
            }
        }
    }
}

WKWatchlistView:
```
        }.onAppear {
            viewModel.fetchWatchlist()
        }

```

2. Run client app and visit watchlist from Settings > Account. Note watchlist values printed out.
3. Toggle off some projects. Confirm filtered watchlist items are printed out.
4. Enter airplane mode and revisit watchlist. Confirm watchlist items are still printed out.
5. Export user library via Help and Feedback > Export. Confirm you can see watchlist response files in the "Watchlists" directory.
